File: PrimeiraEtapa/Semana5/MudaView/MudaView/ViewController.m
//
//  ViewController.m
//  MudaView
//
//  Created by bepid on 10/11/14.
//  Copyright (c) 2014 Bepid. All rights reserved.
//

#import "ViewController.h"
#import "ViewController2.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)mButton:(UIButton *)sender {
    [self performSegueWithIdentifier:@"irTela2" sender:sender];
    
}

-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    if([segue.identifier isEqualToString:@"irTela2"]){
        ViewController2 *view2 = segue.destinationViewController;
        view2.textoRecebido = [NSString stringWithFormat:@"O texto é %@", _mTextField.text];
    }
}


@end

File: Quinta Etapa/AudioDemoTest/audioDemo/ViewController.swift
//
//  ViewController.swift
//  audioDemo
//
//  Created by Diego Vidal on 18/08/15.
//  Copyright (c) 2015 Diego Vidal. All rights reserved.
//

import UIKit
import AVFoundation

class ViewController: UIViewController, AVAudioPlayerDelegate {

    var audioPlayer: AVAudioPlayer?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let url = NSBundle.mainBundle().URLForResource("BackToReality01", withExtension: "mp3")
        var error: NSError?
        
        audioPlayer = AVAudioPlayer(contentsOfURL: url, error: &error)
        
        if let err = error{
            println("audioPlayer Error: \(err.localizedDescription)")
        }
        else{
            audioPlayer?.delegate = self
            audioPlayer?.prepareToPlay()
        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func playAudio(sender: UIButton) {
        if let music = audioPlayer{
            music.play()
        }
    }

    @IBAction func stopAudio(sender: UIButton) {
        if let music = audioPlayer{
            music.stop()
        }
        
    }
    
    @IBAction func adjustVolume(sender: UISlider) {
        if let music = audioPlayer{
            music.volume = sender.value
        }
    }
    
    func audioPlayerDidFinishPlaying(player: AVAudioPlayer!, successfully flag: Bool) {
        println("Encerrou a execução do audio")
    }
}

